# Divide by zero
In Go, dividing an integer by zero leads to a panic, which might interrupt execution of the program and lead to program termination.


## Recommendation
Check that all user inputs used as a divisor are non-zero. It is also possible to handle a panic, generated by division by zero, using the built-in function `recover`.


## Example
The following example shows data received from user input being used as a divisor and possibly causing a divide-by-zero panic.


```go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Printf("Usage: ./program value\n")
		return
	}
	val1 := 1337
	value, _ := strconv.Atoi(os.Args[1])
	out := val1 / value
	fmt.Println(out)
	return
}

```
This can be fixed by testing the divisor against against zero:


```go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Printf("Usage: ./program value\n")
		return
	}
	val1 := 1337
	value, _ := strconv.Atoi(os.Args[1])
	if value == 0 {
		fmt.Println("Division by zero attempted!")
		return
	}
	out := val1 / value
	fmt.Println(out)
	return
}

```
